botocore/client.py

```diff
@@ -248,6 +268,37 @@ def _get_client_args(self, service_model, region_name, is_secure,
             'client_config': new_config
         }
 
+    def _set_s3_presign_signature_verion(
+            self, service_name, endpoint_resolver, client_config,
+            scoped_config, event_emitter, region_name, endpoint_metadata):
+        if service_name != 's3':
+            return
+
+        # This will return the manually configured signature version, or None
+        # if none was manually set. If a customer manually sets the signature
+        # version, we always want to use what they set.
+        provided_signature_version = _get_configured_signature_version(
+            service_name, client_config, scoped_config)
+        if provided_signature_version is not None:
+            return
+
+        # Check to see if the region is a region that we know about. If we
+        # don't know about a region, then we can safely assume it's a new
+        # region that is sigv4 only, since all new S3 regions only allow sigv4.
+        regions = endpoint_resolver.get_available_endpoints(
```

Comment:
We should take a follow up action item to remove this.  This should live in the `endpoints.json` file.  I think it would be a reasonable change to have the default values be s3v4 in the endpoints file and call out all the existing regions that support `s3` explicitly.